### ISSUE TITLE
Error: Array contains no element matching the predicate.

### ISSUE THREAD

**Issue by djthornton1212** [mathze/workflow-dispatch-action#7]
Hello @mathze I'm trying to use your actions, but keep running into an error after the job is triggered.

![image](https://user-images.githubusercontent.com/99346713/156262562-f463b214-0461-4948-8b49-e8b04c48d541.png)

Any thoughts on what I might be doing wrong?

It appears as though it finds the job but unexpectedly fails. Note that the workflow_dispatch does start and successfully run.

Here's my step:
```yaml
 - name: Docker Dispatch
        id: dispatch
        uses: mathze/workflow-dispatch-action@v1
        with:
          owner: company
          repo: github-actions
          token: ${{ steps.generator.outputs.token }}
          workflow-name: docker-deploy.yml
          ref: develop
          payload: ${{ steps.json.outputs.ARTIFACT_JSON }}
          trigger-interval: 10s
          use-marker-step: true
          fail-on-error: true
          wait-interval: 30s
```

**Comment by djthornton1212** [comment]
I suspect the issue to be around here: https://github.com/mathze/workflow-dispatch-action/blob/321be7f6f115c0537870af320cdd0c96f539a106/src/main/kotlin/usecases/WorkflowRuns.kt#L156

I've come to this conclusion from 2 points:
1. The last output `Found 0 removed runs` <=>`logger.info("Found ${removedRuns.size} removed runs")`
2.

**Comment by mathze** [comment]
Hi @djthornton1212 thank you for you report. Currently, I've no access to my PC until tomorrow. But as far as I'm back in town I will have a look at it.

**Comment by djthornton1212** [comment]
Hello @mathze, and thank you for the response. I've just forked the repo to see if I can determine the root cause. If I'm able to determine it I'll submit a PR for you to review.

**Comment by djthornton1212** [comment]
@mathze Ok, I finally figured out what was causing the issue. I'm new to Kotlin (this is my 1st) so it took me longer than I'd like to admit to find the issue then how to correct it. I think the PR opened should address the issue. My jobs are consistently passing now.

I kept trying to figure out how to prevent the run from being entered into the new runs and just couldn't get that figured out. So, I decided to let it in but just filter it out before fetching the jobs.

Let me know if I need to make any changes.
